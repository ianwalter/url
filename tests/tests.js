const { test } = require('@ianwalter/bff')
const Url = require('..')

test('simple url', ({ expect }) => {
  const urlString = 'https://ianwalter.dev/'
  const url = new Url(urlString)
  expect(url.toString()).toBe(urlString)
  expect(url.hostname).toBe('ianwalter.dev')
  expect(url.origin).toBe('https://ianwalter.dev')
  expect(JSON.stringify(url)).toMatchSnapshot()
})

test('url with query', ({ expect }) => {
  const urlString = 'https://ianwalter.dev/search?query=MacBook&page=1'
  const url = new Url(urlString)
  expect(url.toString()).toBe(urlString)
  expect(JSON.stringify(url)).toMatchSnapshot()
})

test('url with base', ({ expect }) => {
  const path = '/spotlight-on-decision-tree'
  const url = new Url(path, 'https://ianwalter.dev')
  expect(url.toString()).toBe(`https://ianwalter.dev${path}`)
  expect(url.pathname).toBe(path)
})

test('URL instance', ({ expect }) => {
  const url = new Url(new URL('https://ianwalter.dev'))
  expect(url).toMatchSnapshot()
})
