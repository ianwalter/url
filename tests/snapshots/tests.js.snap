// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL instance 1`] = `
Url {
  "hash": "",
  "host": "ianwalter.dev",
  "hostname": "ianwalter.dev",
  "href": "https://ianwalter.dev/",
  "origin": "https://ianwalter.dev",
  "password": "",
  "pathname": "/",
  "port": "",
  "protocol": "https:",
  "search": Object {},
  "toString": [Function],
  "username": "",
}
`;

exports[`simple url 1`] = `"{\\"hash\\":\\"\\",\\"host\\":\\"ianwalter.dev\\",\\"hostname\\":\\"ianwalter.dev\\",\\"href\\":\\"https://ianwalter.dev/\\",\\"origin\\":\\"https://ianwalter.dev\\",\\"password\\":\\"\\",\\"pathname\\":\\"/\\",\\"port\\":\\"\\",\\"protocol\\":\\"https:\\",\\"username\\":\\"\\",\\"search\\":{}}"`;

exports[`url with query 1`] = `"{\\"hash\\":\\"\\",\\"host\\":\\"ianwalter.dev\\",\\"hostname\\":\\"ianwalter.dev\\",\\"href\\":\\"https://ianwalter.dev/search?query=MacBook&page=1\\",\\"origin\\":\\"https://ianwalter.dev\\",\\"password\\":\\"\\",\\"pathname\\":\\"/search\\",\\"port\\":\\"\\",\\"protocol\\":\\"https:\\",\\"username\\":\\"\\",\\"search\\":{\\"query\\":\\"MacBook\\",\\"page\\":\\"1\\"}}"`;
